### PR TITLE
Fix: Style for onboarding flow

### LINKED ISSUE #2111: Mobile & Tablet: Colony creation and profile onboarding flows
## Description

The current onboarding (colony and profile creation flows) flow simplified for the beta launch was only prioritized for desktop, so mobile and tablet optimizations need to be completed.

The current flows live on production need to be optimized for mobile and tablet. 

Current mobile: 
image

Current tablet:
image

## Figma
Important: The original designs are included here however, these should only be used for references (UI layout etc) since some of the UI and information has changed, so making the current version optimised is required  - 
https://www.figma.com/file/5V8pr7iMwXsT9L3VAZsmUt/Colony-v5?type=design&node-id=11899-486016&mode=design&t=MCnMNgnf920Kriy7-4

### PR DESCRIPTION
**Pull request by mmioana**
## Description

Refactor landing page item to render children
Fix styling issue on smaller devices

## Testing

TODO: Button styling issue for actionable item list on landing page.

* Step 1. Open landing page
* Step 2. Check the actionable item list on both mobile and desktop

Resolves #2111

### CODE REVIEW

**Review by melyndav — CHANGES_REQUESTED** [review]
Hey @mmioana congrats on getting started with your first PR.

The scope of the issue is to make the landing page, profile creation and colony creation flows responsive for mobile. 

Currently, only desktop was built for these and they are not optimised for our mobile users. Tablet also needs to be considered as part of this scope.

In the issue you'll find a Figma link that shows the complete mobile flows and the landing page variations. Please let me know if you'd like a call to walk through what's needed for this issue. 👍

**Review by melyndav — CHANGES_REQUESTED** [review]
Hey again @mmioana nice start on this one! A big one to tackle. 

Some feedback below to align the UI with the Figma designs and existing mobile interactions: 

### Landing page

1. The padding between the 'Colony app' title and the 'Welcome to Colony' should match Figma.

image


2. The blue pill should be removed on mobile as per Figma

image


3. The icon and background box hsould be aligned to the top of the content block to match the Figma design.

image



### Flow

1. The stepped navigation pills are missing the background styling.

image


2. The buttons should be styled to match Figma.

image

image


3. I have my wallet connected but the wrong header is showing. I should see the chain, my avatar and the settings icon as per the figma designs. 

image

4. Padding between these two select boxes should match Figma.

image


5. These input fields should be aligned according to Figma (stacked for better touch interactions)

image


6. The button buttons should be sticky and the content scrolls.

7. These radios should use the existing component and match the desktop styling

image
image

8. Can you make this button in the bottom widget within the modal full width?

image

**Review by rdig — APPROVED** [review]
Codewise this looks good. I like the desktop / mobile sidebar refactor into a separate component, and the introduction of `enum`s for the flow types.

However, please take care of the styling issues that were pointed out by Mel, before merging this

Screenshot from 2024-06-13 18-53-22

File: src/components/frame/LandingPage/LandingPage.tsx
Comment (mmioana) [reply src/components/frame/LandingPage/LandingPage.tsx:99]: As only one item on the page is enabled and setter is already used, thought to comment the actual state to use it after we enable the other items

File: src/components/v5/shared/Stepper/partials/StepperButton/StepperButton.tsx
Comment (rumzledz) [inline src/components/v5/shared/Stepper/partials/StepperButton/StepperButton.tsx:30]: Great start on your 1st PR @mmioana ! 🥳 

Code-wise all good! This is not a change request but thought I'd say this would be pretty cool as a hook i.e.
```js
const { ref, scrollIntoView } = useScrollIntoView(withWhateverParams)
```

**Review by bassgeta — CHANGES_REQUESTED** [review]
Smashing work on your first PR, immediately a refactor, nice :sunglasses: 
Codewise it looks good, will approve after the UI issues have been resolved!

**Review by melyndav — CHANGES_REQUESTED** [review]
Hey @mmioana,

Nice work on the requested updates! Just one minor change is needed to align it with the Figma designs.

1. This heading doesn't need to be shown and isn't in the designs, so please remove and match the paddings accordingly:

image

Thanks again for your continued work on this! 🐞

**Review by melyndav — APPROVED** [review]
Thanks for the final fix on this PR @mmioana, appreciate your work! 🎉

**Review by bassgeta — APPROVED** [review]
Nice changes, good job handling the myriad of required CSS fixes! :muscle:

**Review by rdig — APPROVED** [review]
Looks good now. Played around with it a bit and doesn't seem to break at any width.

[Screencast from 2024-06-18 23-32-53.webm](https://github.com/JoinColony/colonyCDapp/assets/1193222/a50add34-7a43-469a-9be1-7987e6219f25)

Nicely done.

**Pull request merged**
